src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { MouseEventHandler, useState } from 'react';
+
+import * as RadixCheckbox from '@radix-ui/react-checkbox';
+import { IconCheckBoxCheck } from '../Icons';
+
+import classNames from 'classnames';
+
+import styles from './Checkbox.module.scss';
+
+export interface CheckboxProps {
+  className?: string;
+  text?: string;
+  link?: string;
+  isDisabled?: boolean;
+  checked?: boolean;
+  name: string;
+  hasLightBg?: boolean;
+  onChange?: (value: boolean) => void;
+  redirectLinkTriggered?: () => void;
+}
+
+export const Checkbox = ({
+  isDisabled = false,
+  className,
+  text,
+  link,
+  checked = false,
+  name,
+  hasLightBg =false,
+  onChange,
+  redirectLinkTriggered
+}: CheckboxProps) => {
+  const [isChecked, setIsChecked] = useState(checked);
+
+  const handleOnChange = (checkState: boolean) => {
+    if (!isDisabled) {
+      setIsChecked(checkState);
+      onChange(checkState);
+    }
+  };
+
+  const handleClick = (event: React.MouseEvent<HTMLSpanElement>) => {
+    if (redirectLinkTriggered && !isDisabled) {
+      event.preventDefault();
+      redirectLinkTriggered();
+    }
+  };
+
+  const appendLabelComponent = () => {
+    if (text && link) {
+      return (
+        <label className={classNames(styles.Label)} htmlFor={name}>
+          <span>{text}</span>
+          <span onClick={handleClick} className={classNames(styles.Link, styles.hasLable)}>
+            {link}
+          </span>
+        </label>
+      );
+    } else if (link) {
+      return (
+        <label className={classNames(styles.Label,styles.hasOnlyLink)} htmlFor={name}>
+          <span onClick={handleClick} className={classNames(styles.Link)}>
+            {link}
+          </span>
+        </label>
+      );
+    } else {
+      return (
+        <label className={classNames(styles.Label)} htmlFor={name}>
+          {text}
+        </label>
+      );
+    }
+  };
+
+  return (
+    <div
+      className={classNames(className, styles.Container, `${isDisabled ? styles.DisabledCheckbox : ''}`,
+      `${hasLightBg ? styles.LightBg : ''}`
+      )}
+    >
+      <RadixCheckbox.Root
+        className={classNames(styles.CheckboxRoot, `${isChecked ? styles.CheckedCheckboxRoot : ''}`)}
+        checked={isChecked}
+        disabled={isDisabled}
+        onCheckedChange={(checked: boolean) => handleOnChange(checked)}
+        id={name}
+      >
+        <RadixCheckbox.Indicator className={classNames(styles.CheckboxIndicator)}>
+          <IconCheckBoxCheck width={10} height={8} />
+        </RadixCheckbox.Indicator>
+      </RadixCheckbox.Root>
+      {text || link ? appendLabelComponent() : null}
+    </div>
+  );
+};
```

Comment:
```suggestion
import React, { useState } from 'react';

import * as RadixCheckbox from '@radix-ui/react-checkbox';
import { IconCheckBoxCheck } from '../Icons';

import styles from './Checkbox.module.scss';
import classNames from 'classnames';
const cx = classNames.bind(styles);

export interface CheckboxProps {
  className?: string;
  name: string;
  text?: string;
  link?: string;
  isDisabled?: boolean;
  checked?: boolean;
  hasLightBackground?: boolean;
  onChange?: (value: boolean) => void;
  onLinkClick?: () => void;
}

export const Checkbox = ({
  className,
  name,
  text,
  link,
  isDisabled,
  checked,
  hasLightBackground,
  onChange,
  onLinkClick
}: CheckboxProps) => {
  const [isChecked, setIsChecked] = useState(checked);

  const handleChange = (checkState: boolean) => {
    if (!isDisabled) {
      setIsChecked(checkState);
      onChange(checkState);
    }
  };

  const handleClick = (event: React.MouseEvent<HTMLSpanElement>) => {
    if (onLinkClick && !isDisabled) {
      event.preventDefault();
      onLinkClick();
    }
  };

  const getLabelComponent = () => {
    if (text && link) {
      return (
        <label className={styles.Label} htmlFor={name}>
          <span>{text}</span>
          <span onClick={handleClick} className={classNames(styles.Link, styles.HasLabel)}>
            {link}
          </span>
        </label>
      );
    } else if (link) {
      return (
        <label className={classNames(styles.Label, styles.hasOnlyLink)} htmlFor={name}>
          <span onClick={handleClick} className={styles.Link}>
            {link}
          </span>
        </label>
      );
    } else {
      return (
        <label className={classNames(styles.Label)} htmlFor={name}>
          {text}
        </label>
      );
    }
  };

  return (
    <div className={cx(className, styles.Container, { DisabledCheckbox: isDisabled, LightBg: hasLightBackground })}>
      <RadixCheckbox.Root
        id={name}
        className={cx(styles.CheckboxRoot, { CheckedCheckboxRoot: isChecked })}
        checked={isChecked}
        disabled={isDisabled}
        onCheckedChange={handleChange}
      >
        <RadixCheckbox.Indicator className={styles.CheckboxIndicator}>
          <IconCheckBoxCheck width={10} height={8} />
        </RadixCheckbox.Indicator>
      </RadixCheckbox.Root>
      {(text || link) && getLabelComponent()}
    </div>
  );
};
```

Couple things I would change here:
- We can use `cx` for conditional styling rather than interpolated strings.
- Make sure to use Pascal case for all classnames (styles.hasLable -> styles.HasLabel).
- Preferred naming of function props is onFunctionName (e.g. redirectLinkTriggered -> onLinkClick).

Reply:
Used CX, Used Pascal case for class names, and updated the function names.